projects/kit/components/pager/pager.component.ts

```diff
@@ -0,0 +1,141 @@
+import {NgIf, NgTemplateOutlet} from '@angular/common';
+import type {AfterViewInit, QueryList, TemplateRef} from '@angular/core';
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    inject,
+    Input,
+    signal,
+    ViewChildren,
+} from '@angular/core';
+import {takeUntilDestroyed} from '@angular/core/rxjs-interop';
+import {FormsModule} from '@angular/forms';
+import {
+    WA_INTERSECTION_ROOT,
+    WaIntersectionObservee,
+    WaIntersectionObserverDirective,
+} from '@ng-web-apis/intersection-observer';
+import {
+    MutationObserverService,
+    WA_MUTATION_OBSERVER_INIT,
+} from '@ng-web-apis/mutation-observer';
+import {TuiRepeatTimes} from '@taiga-ui/cdk/directives/repeat-times';
+import {tuiProvide} from '@taiga-ui/cdk/utils/miscellaneous';
+import {delay} from 'rxjs';
+
+@Component({
+    standalone: true,
+    selector: 'tui-pager',
+    imports: [
+        FormsModule,
+        NgIf,
+        NgTemplateOutlet,
+        TuiRepeatTimes,
+        WaIntersectionObservee,
+        WaIntersectionObserverDirective,
+    ],
+    templateUrl: './pager.component.html',
+    styleUrls: ['./pager.styles.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        tuiProvide(WA_INTERSECTION_ROOT, ElementRef),
+        MutationObserverService,
+        {
+            provide: WA_MUTATION_OBSERVER_INIT,
+            useValue: {
+                attributeOldValue: true,
+                characterData: true,
+                childList: true,
+                subtree: true,
+            },
+        },
+    ],
+    host: {
+        '[style.--t-gap.px]': 'gap',
+        '[style.max-width.px]': 'maxWidth()',
+    },
+})
+export class TuiPager implements AfterViewInit {
+    private readonly cd = inject(ChangeDetectorRef);
+
+    @ViewChildren('item')
+    protected children?: QueryList<ElementRef<HTMLElement>>;
+
+    protected visible = new Set<number>();
+    protected maxWidth = signal(0);
+    protected left = signal(0);
+
+    protected readonly $ = inject(MutationObserverService, {self: true})
+        .pipe(delay(0), takeUntilDestroyed())
+        .subscribe(() => this.calculateVisibleWidth());
+
+    @Input()
+    public max = 6;
+
+    @Input()
+    public gap = 8;
+
+    @Input()
+    public count = this.max;
+
+    @Input()
+    public template?: TemplateRef<unknown>;
+
+    public activeItemIndex = 0;
+
+    @Input('activeItemIndex')
+    public set itemIndex(activeItemIndex: number) {
+        this.activeItemIndex = activeItemIndex;
+        this.move();
+    }
+
+    public ngAfterViewInit(): void {
```

Comment:
I think we can get rid of it once we move from setter to ngOnChange

Reply:
fixed